homeassistant/components/flux_led/light.py

```diff
@@ -171,133 +141,231 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
-                CONF_EFFECT_SPEED, DEFAULT_EFFECT_SPEED
-            ),
+
+    async def async_new_lights(bulbs: dict):
+        """Add new bulbs when they are found or configured."""
+        bulb_coordinator = (
+            hass.data[DOMAIN].get(entry.entry_id, {}).get(BULB_COORDINATOR)
         )
+        lights = []
+
+        for bulb_id, bulb_details in bulbs.items():
+            effect_speed = entry.options.get(bulb_id, {}).get(
+                CONF_EFFECT_SPEED,
+                entry.options.get("global", {}).get(
+                    CONF_EFFECT_SPEED, DEFAULT_EFFECT_SPEED
+                ),
+            )
+
+            coordinator = FluxLEDCoordinator(
+                hass=hass,
+                name=bulb_details[CONF_NAME],
+                update_interval=DEFAULT_SCAN_INTERVAL,
+                ip_address=bulb_details[CONF_HOST],
+                scan_coordinator=bulb_coordinator,
+            )
+
+            await coordinator.async_refresh()
 
-        light = FluxLight(add_device)
-        lights.append(light)
+            lights.append(
+                FluxLight(
+                    coordinator=coordinator,
+                    unique_id=bulb_id,
+                    device=bulb_details,
+                    effect_speed=effect_speed,
+                    hass=hass,
+                )
+            )
+
+        async_add_entities(lights)
 
-    async_add_entities(lights)
+    await async_new_lights(entry.data[CONF_DEVICES])
 
+    async_dispatcher_connect(hass, SIGNAL_ADD_DEVICE, async_new_lights)
 
-class FluxLight(LightEntity):
-    """Representation of a Flux light."""
 
-    def __init__(self, device):
-        """Initialize the light."""
-        self._name = device["name"]
-        self._ipaddr = device[CONF_HOST]
-        self._protocol = device[CONF_PROTOCOL]
-        self._mode = device[ATTR_MODE]
-        self._custom_effect = device[CONF_CUSTOM_EFFECT]
-        self._effect_speed = device[CONF_EFFECT_SPEED]
-        self._bulb = None
-        self._error_reported = False
+class FluxLEDCoordinator(DataUpdateCoordinator):
+    """Update Coordinator for a specific light entity."""
 
-    def _connect(self):
-        """Connect to Flux light."""
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        name: str,
+        update_interval: int,
+        ip_address: str,
+        scan_coordinator: FluxLEDListUpdateCoordinator = None,
+    ):
+        """Initialize the update coordinator."""
+
+        super().__init__(
+            hass=hass,
+            logger=_LOGGER,
+            name=name,
+            update_interval=timedelta(milliseconds=update_interval),
+        )
 
-        self._bulb = WifiLedBulb(self._ipaddr, timeout=5)
-        if self._protocol:
-            self._bulb.setProtocol(self._protocol)
+        self._ip_address = ip_address
+        self._name = name
+        self.scan_coordinator = scan_coordinator
+
+        self.light = WifiLedBulb(self._ip_address)
```

Comment:
We're not using all these attributes.
```suggestion
        self.light = WifiLedBulb(ip_address)
```

Reply:
OK updated.